source/base/function_lib.cc

```diff
@@ -2483,6 +2483,130 @@ namespace Functions
     return interpolate (data_values, ix, p_unit);
   }
 
+  /* ---------------------- Polynomial ----------------------- */
+
+
+
+  template <int dim>
+  Polynomial<dim>::
+  Polynomial(const Table<2,double> &exponents,
+             const Vector<double>   &coefficients,
+             const unsigned int   n_components)
+    :
+    Function<dim> (n_components),
+    exponents (exponents),
+    coefficients(coefficients)
+  {
+    Assert(exponents.n_rows() == coefficients.size(),
+           ExcDimensionMismatch(exponents.n_rows(), coefficients.size()));
```

Comment:
I prefer to check exponents.n_cols() <=dim, so the user can feed the constructor with a table with 1 column for 1d polynomials and a table with 2 columns for 2d polynomials. This may allow a good reduction in memory consumption for cases where large number of 1d or 2d monomials are used.